src/compiler/types.ts

```diff
@@ -643,21 +644,23 @@ namespace ts {
         initializer?: Expression;           // Optional initializer
     }
 
-    export interface ObjectLiteralElement extends Declaration {
+    export interface ObjectLiteralElementLike extends Declaration {
         _objectLiteralBrandBrand: any;
         name?: PropertyName;
    }
 
+    export type ObjectLiteralElement = PropertyAssignment | ShorthandPropertyAssignment | SpreadObjectLiteralAssignment | MethodDeclaration | AccessorDeclaration;

```

Comment:
From our in-person discussion, this is to capture JSX-based types later.
Also, we decided to swap the -Like/no-Like names.